picard/formats/id3.py

```diff
@@ -494,6 +494,19 @@ def _get_tags(self, filename):
         except mutagen.id3.ID3NoHeaderError:
             return compatid3.CompatID3()
 
+    def _load_preserved_config(self, metadata):
+        # Adding additional config to metadata to prevent unecessary saving
+        metadata['~config:rating_steps'] = config.setting['rating_steps']
+        metadata['~config:rating_user_email'] = config.setting['rating_user_email']
```

Comment:
I have a feeling here that we are over complicating this solution by trying to make it perfect for all combinations of changing option settings and saving combinations of files. e.g. if we change options and save one file and change them back again before saving the rest. This seems to be why we are saving configuration settings in file metadata.

Most people will have a standard set of settings that they use daily - and won't change them frequently. The easiest solution in my opinion is to preserve globally the settings when Picard is started, and if current settings have changed from these, then we save the files. If we want to get a bit cleverer, then we can have a list of settings that we check or ignore (so that e.g. script settings changed don't trigger global saves because script changes of note will have changed metadata which will trigger a save).